optuna/integration/dask.py

```diff
@@ -319,7 +319,7 @@ def set_trial_intermediate_value(
         storage_name: str,
         trial_id: int,
         step: int,
-        intermediate_value: float,
+        intermediate_value: Union[float, Sequence[float]],
```

Comment:
Instead of using `Union`, it is more modern to use `|`, and Optuna is gradually adopting it. Could you use `|` in this PR as well? Additionally, instead of the `typing` version, could you use `collections.abc` for `Sequence`?

```suggestion
        intermediate_value: float | Sequence[float],
```

Reply:
This comment will be applied to other files.